extensions/amp-story-shopping/0.1/test/test-amp-story-shopping-config.js

```diff
@@ -231,5 +202,91 @@ describes.realWin(
+      expect(spy).to.have.been.calledWith(
+        'AMP-STORY-SHOPPING-CONFIG',
+        errorString
+      );
+    });
+
+    it('should fail config validation because an expected string JSON value is not a valid HTML id', async () => {
+      const invalidConfig = JSON.parse(JSON.stringify(defaultInlineConfig));
+      const invalidValue = '1234city-pop'; // productId starts with a number, so it is an invalid HTML Id
+      invalidConfig['items'][0]['productId'] = invalidValue;
+
+      const errorString = `Error: productId ${invalidValue} is not a valid HTML Id`;
+      const spy = env.sandbox.spy(user(), 'warn');
+      await createAmpStoryShoppingConfig(null, invalidConfig);
+      expect(spy).to.have.been.calledWith(
+        'AMP-STORY-SHOPPING-CONFIG',
+        errorString
+      );
+    });
+
+    it('should fail config validation because an expected number JSON value is not a valid number', async () => {
+      const invalidConfig = JSON.parse(JSON.stringify(defaultInlineConfig));
+      const invalidValue = 'two dozen watermelons'; // two dozen watermelons is not an actual price.
+      invalidConfig['items'][0]['productPrice'] = invalidValue;
+
+      const errorString = `Error: Value ${invalidValue} for field productPrice is not a number`;
+      const spy = env.sandbox.spy(user(), 'warn');
+      await createAmpStoryShoppingConfig(null, invalidConfig);
+      expect(spy).to.have.been.calledWith(
+        'AMP-STORY-SHOPPING-CONFIG',
+        errorString
+      );
+    });
+
+    it('should fail config validation because an expected string JSON value is not a valid currency code symbol', async () => {
+      const invalidConfig = JSON.parse(JSON.stringify(defaultInlineConfig));
+      const invalidValue = 'ZABAN'; // This is not a valid currency symbol code
+      invalidConfig['items'][0]['productPriceCurrency'] = invalidValue;
+
+      const errorString = `Error: productPriceCurrency ${invalidValue} is not a valid currency code`;
+      const spy = env.sandbox.spy(user(), 'warn');
+      await createAmpStoryShoppingConfig(null, invalidConfig);
+      expect(spy).to.have.been.calledWith(
+        'AMP-STORY-SHOPPING-CONFIG',
+        errorString
+      );
+    });
+
+    it('should fail config validation because an expected string JSON value is not a valid url', async () => {
+      const invalidConfig = JSON.parse(JSON.stringify(defaultInlineConfig));
+      const invalidValue = 'http://zapp'; // This is not a valid url
+      invalidConfig['items'][0]['productUrl'] = invalidValue;
+      const spy = env.sandbox.spy(url, 'assertHttpsUrl');
+      const keyedShoppingConfig = await createAmpStoryShoppingConfig(
+        null,
+        invalidConfig
+      );
+      expect(spy).to.have.been.calledWith(
+        `${invalidValue}`,
+        'amp-story-shopping-config productUrl'
+      );
+      expect(Object.keys(keyedShoppingConfig).length).to.eql(1);
+      expect(Object.keys(keyedShoppingConfig)[0]).to.eql('art');
```

Comment:
Mind also adding these 2 expectations to the other tests above? Specifically for the "test validate required fields for config" test and all the tests below it